cmd/mysterium_client/command_run/command.go

```diff
@@ -5,90 +5,94 @@ import (
 	"github.com/mysterium/node/bytescount_client"
 	"github.com/mysterium/node/client_connection"
 	"github.com/mysterium/node/communication"
-    "github.com/mysterium/node/identity"
+	"github.com/mysterium/node/communication/nats_dialog"
+	"github.com/mysterium/node/communication/nats_discovery"
+	"github.com/mysterium/node/identity"
 	"github.com/mysterium/node/openvpn"
-	vpn_session "github.com/mysterium/node/openvpn/session"
+	"github.com/mysterium/node/openvpn/session"
 	"github.com/mysterium/node/server"
+	"github.com/mysterium/node/server/dto"
 	"github.com/mysterium/node/tequilapi"
 	"github.com/mysterium/node/tequilapi/endpoints"
+	"path/filepath"
 	"time"
 )
 
 type CommandRun struct {
+	//TODO this must disappear or become a private field
 	MysteriumClient server.Client
 
-	DialogEstablisherFactory func(identity identity.Identity) communication.DialogEstablisher
-	dialog                   communication.Dialog
+	identityManager identity.IdentityManagerInterface
```

Comment:
Very strange to inject all those variables, just to use them for reinjecting in `Run`

Reply:
So your idea is that we should construct everything in factory method, and leave Run method to start serving tequilapi only (or do any other startup actions needed) ?

Reply:
Refactored.